gc_tests/tests/cycle_in_non_gc_value_object_graph.rs

```diff
@@ -0,0 +1,35 @@
+// Run-time:
+//  status: success
+
+extern crate gcmalloc;
+
+use gcmalloc::{collect, Debug, DebugFlags, Gc};
+
+struct S {
+    x: usize,
+    y: Gc<usize>,
+}
+
+fn main() {
+    gcmalloc::debug_flags(DebugFlags::new().sweep_phase(false));
+
+    let mut s = Box::new(S {
+        x: 0,
+        y: Gc::new(123),
+    });
+
+    // We take a reference to a non-garbage-collected heap object and place it
+    // inside its own heap contents with the knowledge that the collector will
+    // look through its fields and enqueue them for marking.
+    //
+    // If the original object hasn't been marked as "seen" before any of its
+    // fields are popped off the queue and processed then we can end up in an
+    // infinite loop.
+    s.x = s.as_ref() as *const _ as usize;
+
+    gcmalloc::collect();
+
+    // Until I solve the halting problem, this takes advantage of lang-testers
+    // time-out. If we got here, then the test passed.
+    assert!(true)
```

Comment:
Can't you kill this last assertion?

I'd just have a comment like: `If the GC gets stuck in a cycle, we won't get here`.

Reply:
It was just there to hammer-home the point really. I've no strong feelings on it though, if you think it's unnecessary I'll remove it.

Reply:
I think it'd be clearer to just have a short comment.

Reply:
Fixed: 54cbbe987b36ea558da8f3ed59a20096a1adf53b